### ISSUE TITLE
API for Custom User Defined Error Handling Logic Inside RabbitMQ

### ISSUE THREAD

**Issue by WonderPanda** [golevelup/nestjs#127]
Instead of adding a new error handling behavior at the connection level what if we added the option to supply your own error handling logic at the decorator level so that people can flexibly handle error cases differently across different queues/exchanges etc.

I'm wondering about adding in a new option method to the decorator that might look something like this:

```
errorHandler?: (
      connection: AmqpConnection,
      channel: amqplib.Channel,
      msg: amqplib.ConsumeMessage
    ) => void | Promise<void> 
```

That way if an error happens and an errorHandler was provided the current connection, channel and message that caused the error will be provided to your error handler which can do whatever it wants with this information. That would allow the community to supply additional plugins that cover difference scenarios and desired behavior without needing to bloat the core AmqpConnection code with more and more Error Handling Behaviors

**Comment by WonderPanda** [comment]
Instead of adding a new error handling behavior at the connection level what if we added the option to supply your own error handling logic at the decorator level so that people can flexibly handle error cases differently across different queues/exchanges etc.

I'm wondering about adding in a new option method to the decorator that might look something like this:
```
errorHandler?: (
      connection: AmqpConnection,
      channel: amqplib.Channel,
      msg: amqplib.ConsumeMessage
    ) => void | Promise<void> 
```
That way if an error happens and an errorHandler was provided the current connection, channel and message that caused the error will be provided to your error handler which can do whatever it wants with this information. That would allow the community to supply additional plugins that cover difference scenarios and desired behavior without needing to bloat the core AmqpConnection code with more and more Error Handling Behaviors

**Comment by WonderPanda** [comment]
Closed by #126
